src/openassetio-core/include/openassetio/specifications.hpp

```diff
@@ -0,0 +1,137 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2013-2022 The Foundry Visionmongers Ltd
+#pragma once
+
+#include <memory>
+#include <string>
+#include <unordered_map>
```

Comment:
Capturing off-line discussion:

Work out strategy to make sure we don't leak the storage implementation.

Reply:
Used pimpl in the rewrite in 01d6a8ea